### PR TITLE
Adding “Web Development with AMP” Courses to amp.dev

### PR DESCRIPTION
**Pull request by mandarini**
We’re developers working with Ben Morss (@morsssss ) to develop content for the amp.dev site. We’re adding the “Web Development with AMP” series which is comprised of three courses (Beginner, Intermediate, and Advanced). We know that the AMP team is still deciding where these courses should reside in the navigation, but the purpose of this PR is to start the review of the content itself and make any necessary changes.

### CODE REVIEW

**Comment by matthiasrohmer** [comment]
Forwarding this to the core team for a review with regards to content. I am happy to help with integrating this in any of the navigations then 🙂

File: pages/content/amp-dev/documentation/guides-and-tutorials/courses/intermediate-course/accepting-user-input-and-displaying-output.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+- If the form submission fails, an error should be displayed to the user.
+
+- If the form submission succeeds, a welcome message with the user’s name should be displayed.
+
+In a non-AMP page, we might reach for a basic HTML form element to collect our users’ information. AMP provides an [`<amp-form>`]({{g.doc('/content/amp-dev/documentation/components/reference/amp-form.md', locale=doc.locale).url.path}}) component, but this component actually uses the element name "form" as well. So, what’s the difference?
+
+In this section, we’ll explore what a traditional website form is, how AMP forms differ from traditional forms, and how to handle the response from the server when submitting AMP forms.
+
+## Introduction To Forms
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/courses/intermediate-course/accepting-user-input-and-displaying-output.md]: Is it intentional that AMP CORS is not mentioned in this part of the course? Without, the form sample wouldn't work if served from an AMP cache.

Comment (morsssss) [reply pages/content/amp-dev/documentation/guides-and-tutorials/courses/intermediate-course/accepting-user-input-and-displaying-output.md]: Good point! We mention CORS in the next course, but perhaps we should move that to this one.
We don't explain it in detail in any case - we just refer the more advanced student to appropriate docs.

**Comment by morsssss** [comment]
Thanks for submitting this, @mandarini !

Note to @sebastianbenz , @CrystalOnScript, @pbakaus, etc: Before we merge this, I need to figure out with you all where it should really go on the site. We also may wish to resolve some styling issues and add a couple more things.

In other words, expect some changes within the next week or so... and then we'd love to get this on the site in early April.

**Comment by morsssss** [comment]
### It's time to decide: where should we put these courses?

Ideally, I'd like there to be a landing page with brief text summarizing the course content. This would link to two pages: one for teachers, and another for students.

The teacher page would have some brief text as well, then links from which the course materials can be downloaded, as well as links where everything can be viewed in Drive.

The student page would have some brief text, then links to the three courses on the site. Once videos are done, I think we'd want to embed the first one here.

To make all this make sense, I think it's most logical to create a new submenu under "Documentation" called "Courses". This would let us create the Courses environment I just described.

If others think that's one submenu too many, perhaps "Courses" should stay in this section, but it could be under "LEARN" instead of at the bottom. (One could also argue that they should be under "LEARN", but they could be under "START" too, so I feel like they should get their own sidebar area.)

WDYT?

**Comment by morsssss** [comment]
@sebastianbenz , @CrystalOnScript, @pbakaus: wdyt?

**Comment by matthiasrohmer** [comment]
Not sure if this is known to all, therefore thought I'd share it again: plans are that Documentation is not clickable anymore but will always open the flyout and will get a new item "Get Started" which will lead to what's currently the Guides and Tutorial landing pages. This will then be replaced by an overview page similar to the one the components and examples have.

Therefore in my opinion adding another "Courses" item to the flyout seems like a good idea. This could then lead to a dedicated Courses landing page with it's own sidebar. 🙂

**Comment by pbakaus** [comment]
+1 to what @matthiasrohmer said!

**Comment by morsssss** [comment]
Sounds great!

If so,  it looks like I can proceed more or less in the way I proposed... adding a new element to the flyout in the Blueprint, moving all the files to a new directory, creating the landing page, etc. And I'll look at how to create the sidebar 😃 

Is that right?

**Comment by pbakaus** [comment]
Sounds about right to me!

**Comment by matthiasrohmer** [comment]
@morsssss, feel free to reach out on Slack if I can support with setting things up 🙂

**Comment by morsssss** [comment]
This has now been moved to a feature branch on the main repo....

**Comment by matthiasrohmer** [comment]
Closed in favor of #1887.

**Pull request closed without merging**